### PR TITLE
[MIG] pos_loyalty: Migrated into new api

### PR DESCRIPTION
**Pull request by dbu-odoo**
<b>Task:</b> https://www.odoo.com/web#id=13731&view_type=form&model=project.task&action=327
<b>Pad:</b> https://pad.odoo.com/p/r.NmkjQTXj345Cnydy

### CODE REVIEW

File: addons/pos_loyalty/__openerp__.py [hunk 1]
```diff
@@ -20,13 +20,13 @@
...
-        'views/templates.xml'
+        'views/pos_loyalty_templates.xml'
     ],
     'qweb': ['static/src/xml/loyalty.xml'],
     'demo': [
-        'loyalty_demo.xml',
+        'data/pos_loyalty_demo.xml',
     ],
     'installable': True,
     'auto_install': False,
```
Comment (rgo-odoo) [inline addons/pos_loyalty/__openerp__.py]: you can omit this

File: addons/pos_loyalty/__openerp__.py [hunk 2]
```diff
@@ -20,13 +20,13 @@
     'author': 'OpenERP SA',
```
Comment (rgo-odoo) [inline addons/pos_loyalty/__openerp__.py]: OpenERP SA -->> Odoo S.A.

File: addons/pos_loyalty/models/pos_loyalty_reward.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            else:
+                return True
+
+    @api.constrains('reward_type', 'discount_product_id')
+    def _check_discount_product(self):
+        for reward in self:
+            if reward.reward_type == 'discount' and not reward.discount_product_id:
+                raise ValidationError(_('The discount product field is mandatory for discount rewards'))
+            else:
+                return True
```
Comment (rgo-odoo) [inline addons/pos_loyalty/models/pos_loyalty_reward.py]: You can write constrains methods like,
@api.one
@api.constrains('reward_type', 'discount_product_id')
def _check_discount_product(self):
     if self.reward_type == 'discount' and not self.discount_product_id:
         raise ValidationError(_('The "Discount Product" field is mandatory for discount rewards.'))
     return True

Would you make similar changes for all your constrains methods.

File: addons/pos_loyalty/models/pos_order.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+    @api.model
+    def _order_fields(self, ui_order):
+        fields = super(PosOrder, self)._order_fields(ui_order)
+        fields['loyalty_points'] = ui_order.get('loyalty_points', 0)
+        return fields
+
+    @api.model
+    def create_from_ui(self, orders):
+        order_id = super(PosOrder, self).create_from_ui(orders)
```
Comment (rgo-odoo) [inline addons/pos_loyalty/models/pos_order.py]: it should be order_ids

File: addons/pos_loyalty/__openerp__.py (hunk 1 above)
Comment (dbu-odoo) [inline addons/pos_loyalty/__openerp__.py]: ok

File: addons/pos_loyalty/models/pos_loyalty_reward.py (hunk 3 above)
Comment (dbu-odoo) [inline addons/pos_loyalty/models/pos_loyalty_reward.py]: ok done

File: addons/pos_loyalty/models/pos_order.py (hunk 4 above)
Comment (dbu-odoo) [inline addons/pos_loyalty/models/pos_order.py]: ok done

File: addons/pos_loyalty/__openerp__.py (hunk 2 above)
Comment (dbu-odoo) [inline addons/pos_loyalty/__openerp__.py]: According to https://github.com/odoo/odoo/commit/0decfd94d13fb49fe9a1e92f8aacd83a0cddf0b7 commit I remove author.

**Comment by aymenrahmaniisims** [comment]
loyality for Odoo 8 
there i any module for this

**Pull request closed without merging**